GetIntrinsic.js

```diff
@@ -199,18 +218,46 @@ module.exports = function GetIntrinsic(name, allowMissing) {
 	}
 
 	var parts = stringToPath(name);
+	var intrinsicBaseName = parts.length > 0 ? parts[0] : '';
+
+	var intrinsic = getBaseIntrinsic('%' + intrinsicBaseName + '%', allowMissing);
+	var intrinsicRealName = intrinsic.name;
+	var value = intrinsic.value;
+	var skipFurtherCaching = false;
+
+	var alias = intrinsic.alias;
+	if (alias) {
+		intrinsicBaseName = alias[0];
+		$spliceApply(parts, $concat([0, 1], alias));
+	}
 
-	var value = getBaseIntrinsic('%' + (parts.length > 0 ? parts[0] : '') + '%', allowMissing);
-	for (var i = 1; i < parts.length; i += 1) {
-		if (value != null) {
+	for (var i = 1, isOwn = true; i < parts.length; i += 1) {
+		var part = parts[i];
+		if (part === 'constructor' || !isOwn) {
+			skipFurtherCaching = true;
+		}
```

Comment:
can you elaborate on this one?

Reply:
This is so that we don’t needlessly cache properties that cyclically walk up the constructor chain (e.g.:`%Object.prototype.constructor.prototype.constructor.getOwnPropertyDescriptor%`), same with non‑own properties (e.g.:`%Function.call%`).

Reply:
I'm not sure anyone's using `Function.call` in this API; i'd consider that a bugfix.